snowplow_tracker/tracker.py

```diff
@@ -687,29 +692,32 @@ def track_ecommerce_transaction(
         )
         non_empty_string(order_id)
 
-        pb = payload.Payload()
-        pb.add("e", "tr")
-        pb.add("tr_id", order_id)
-        pb.add("tr_tt", total_value)
-        pb.add("tr_af", affiliation)
-        pb.add("tr_tx", tax_value)
-        pb.add("tr_sh", shipping)
-        pb.add("tr_ci", city)
-        pb.add("tr_st", state)
-        pb.add("tr_co", country)
-        pb.add("tr_cu", currency)
+        event = Event()
+        event.pb.add("e", "tr")
+        event.pb.add("tr_id", order_id)
+        event.pb.add("tr_tt", total_value)
+        event.pb.add("tr_af", affiliation)
+        event.pb.add("tr_tx", tax_value)
+        event.pb.add("tr_sh", shipping)
+        event.pb.add("tr_ci", city)
+        event.pb.add("tr_st", state)
+        event.pb.add("tr_co", country)
+        event.pb.add("tr_cu", currency)
 
         tstamp = Tracker.get_timestamp(tstamp)
 
-        self.complete_payload(pb, context, tstamp, event_subject)
+        self.track(
+            event=event, event_subject=event_subject, context=context, tstamp=tstamp
+        )
 
         if items is None:
             items = []
         for item in items:
-            item["tstamp"] = tstamp
-            item["event_subject"] = event_subject
             item["order_id"] = order_id
             item["currency"] = currency
+            item["tstamp"] = tstamp
+            item["event_subject"] = event_subject
+            item["context"] = context
             self.track_ecommerce_transaction_item(**item)
 
         return self
```

Comment:
Can't comment at that point so adding the comment here – can we also add an event class for the mobile screen view? We have that in most trackers. We can just call it `ScreenView` (without mobile) as there won't be any conflict with the older screen view which doesn't need to have it's own event class.